python_modules/dagster/dagster/_config/structured_config/typing_utils.py

```diff
@@ -49,7 +51,7 @@ def set_actual_types_for_type_checking(
         )
 
 
-@dataclass_transform()
+@dataclass_transform(kw_only_default=True, field_specifiers=(Field, FieldInfo))
```

Comment:
https://github.com/pydantic/pydantic/blob/main/pydantic/main.py#L72

Reply:
`kw_only_default` helps with type annotation here, already the new config/resources will break if you feed them non-kwargs (hence the change in tests below)

Reply:
more out of curiosity, what does the field_specifiers do?

Reply:
They basically tell Python & tools like the type checkers what classes to look for for metadata on each field. In this case, it's telling Pyright to look at Pydantic `Field` for metadata like the alias

https://peps.python.org/pep-0681/#field-specifiers has more info:
> `alias` is an optional str parameter that provides an alternative name for the field. This alternative name is used in the synthesized `__init__ `method.

Reply:
FYI this is the declaration in pydantic itself: 

https://github.com/pydantic/pydantic/blob/251c22c7d224d9ed9aea67777a2494674c5d0886/pydantic/main.py#L73

```
@typing_extensions.dataclass_transform(kw_only_default=True, field_specifiers=(Field,))
class ModelMetaclass(ABCMeta):
    def __new__(
        mcs,
        cls_name: str,
        bases: tuple[type[Any], ...],
        namespace: dict[str, Any],
        __pydantic_generic_metadata__: _generics.PydanticGenericMetadata | None = None,
        __pydantic_reset_parent_namespace__: bool = True,
        **kwargs: Any,
    ) -> type:
```

Reply:
Hmm, yeah, looks like they got rid of `FieldInfo`. I think we can remove it as well since it's an internal API rather than something users would use.